WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationForm.swift

```diff
@@ -169,20 +165,25 @@ struct BlazeCampaignCreationForm: View {
             isShowingAISuggestionsErrorAlert = newValue == .failedToLoadAISuggestions
         }
         .alert(isPresented: $isShowingAISuggestionsErrorAlert, content: {
-            Alert(title: Text(Localization.ErrorAlert.title),
-                  message: Text(Localization.ErrorAlert.ErrorMessage.fetchingAISuggestions),
-                  primaryButton: .default(Text(Localization.ErrorAlert.retry), action: {
+            Alert(title: Text(Localization.AISuggestionsErrorAlert.title),
+                  message: Text(Localization.AISuggestionsErrorAlert.ErrorMessage.fetchingAISuggestions),
+                  primaryButton: .default(Text(Localization.AISuggestionsErrorAlert.retry), action: {
                 Task {
                     await viewModel.loadAISuggestions()
                 }
             }),
                   secondaryButton: .cancel())
         })
-        .task {
-            await viewModel.loadAISuggestions()
+        .alert(isPresented: $viewModel.isShowingMissingImageErrorAlert, content: {
+            Alert(title: Text(Localization.NoImageErrorAlert.noImageFound),
+                  dismissButton: .default(Text(Localization.NoImageErrorAlert.ok)))
+        })
+        .onAppear() {
+            viewModel.onAppear()
         }
-        .task {
-            await viewModel.downloadProductImage()
+        LazyNavigationLink(destination: BlazeConfirmPaymentView(viewModel: viewModel.confirmPaymentViewModel),
+                           isActive: $viewModel.isShowingPaymentInfo) {
+            EmptyView()
         }
```

Comment:
Just wondering: does this help with the loading when navigating back? I thought it was nicer to include the navigation link in the button, as using `EmptyView` seems like a workaround.

Reply:
Keeping the ` LazyNavigationLink` inside `Button` label doesn't call the `action` and instead navigates to the confirm payment screen on tap. i.e. I can't check the condition to show an alert; instead, the navigation always happens.

```swift
                Button {
                    viewModel.didTapConfirmDetails()
                } label: {
                    LazyNavigationLink(destination: BlazeConfirmPaymentView(viewModel: viewModel.confirmPaymentViewModel),
                                       isActive: $viewModel.isShowingPaymentInfo) {
                        Text(Localization.confirmDetails)
                    }
                }
```

So, I moved the `LazyNavigationLink` out of the Button. Please share if you have any other solution for this.

Reply:
Thanks for the explanation - this makes sense!